File: jtransc-rt-core/src/com/jtransc/simd/MutableFloat32x4Utils.java
package com.jtransc.simd;

import com.jtransc.annotation.JTranscInvisible;
import com.jtransc.annotation.JTranscSync;

@JTranscInvisible
final public class MutableFloat32x4Utils {
	static private final MutableFloat32x4 temp = MutableFloat32x4.create();

	@JTranscSync
	static public float getAddMul(MutableFloat32x4 l, MutableFloat32x4 r) {
		temp.setToMul(l, r);
		return temp.getSumAll();
	}

	@JTranscSync
	static public String toStringInternal(MutableFloat32x4 v) {
		return "Simd.MutableFloat32x4(" + v.getX() + ", " + v.getY() + ", " + v.getZ() + ", " + v.getW() + ")";
	}
}

File: benchmark_kotlin_mpp/src/commonMain/kotlin/simd/MutableFloat32x4Utils.kt
package simd

object MutableFloat32x4Utils {
	private val temp: MutableFloat32x4 = MutableFloat32x4.create()

	fun getAddMul(l: MutableFloat32x4, r: MutableFloat32x4): Float {
		temp.setToMul(l, r)
		return temp.sumAll
	}


	fun toStringInternal(v: MutableFloat32x4): String {
		return "Simd.MutableFloat32x4(" + v.x.toString() + ", " + v.y.toString() + ", " + v.z
			.toString() + ", " + v.w.toString() + ")"
	}
}